docs/content/guides/developer/getting-started/data-serving.mdx

```diff
@@ -0,0 +1,115 @@
+
+This document outlines the interfaces that are currently available to access the Sui network data, along with a overview of how that's gradually evolving for a great deveoper experience.
+
+## Current data access interfaces
+
+You can use either of the following mechanisms to access Sui network data currently:
+
+- Directly connect to [JSON-RPC](/references/sui-api.mdx) hosted by a Sui [Full node](guides/operator/sui-full-node.mdx) where the [Mainnet](https://fullnode.mainnet.sui.io:443) or [Testnet](https://fullnode.testnet.sui.io:443) or [Devnet](https://fullnode.devnet.sui.io:443) load balancer URLs abstract Mysten Labs managed Full nodes, OR
+- Set up your own [custom indexer](/guides/developer/advanced/custom-indexer.mdx) to continuously load the data of interest in a Postgres database, OR 
+- Utilize a [RPC provider](https://sui.io/developers#dev-tools) (filter by `RPC`) or [Data indexer operator](https://github.com/sui-foundation/awesome-sui?tab=readme-ov-file#indexers--data-services) who have setup their own Full nodes or indexing infrastructure.
+
+:::info
+
+You can use one of the [future-oriented interfaces]((#future-state-data-interfaces)) that are available in alpha or beta, but you must validate the relevant functionality, performance, and cost characteristics for your application before deciding to use those in production.
+
+:::
+
+### JSON-RPC
+
+You can currently get real-time or historical data from a Sui Full node. Retention period for historical data depends on the [pruning strategy](/guides/operator/data-management.mdx#sui-full-node-pruning-policies) implemented by a node operator, though currently all Full nodes are by default configured to implicitly fall back on a [Mysten Labs managed key-value store for historical transaction data](/guides/operator/data-management.mdx#sui-full-node-key-value-store-backup).
+
+:::caution
+
+Websockets based JSON RPCs `suix_subscribeEvent` and `suix_subscribeTransaction` were deprecated in July 2024. You should not rely on those RPCs in your applications. Refer to [this section](#future-state-data-interfaces) to learn about a future alternative.
+
+:::
+
+### Custom indexer
+
+If you need object version history or another type of historical data where you may not fully rely on a Full node or RPC provider, or if you have specific query patterns that could be best served from a relational database, then you can setup your own [custom indexer](/guides/developer/advanced/custom-indexer.mdx) or reach out to a [Data indexer operator](https://github.com/sui-foundation/awesome-sui?tab=readme-ov-file#indexers--data-services) who may already have setup one. 
+
+If you setup your own indexer, you are responsible for its ongoing maintenance and the related infrastructure & operational costs. You can reduce your costs by implementing a pruning strategy for the relational database by taking into account the retention needs of your application.
+
+## Future data access interfaces
```

Comment:
We talked about splitting the indexing framework out into its own category, sitting between gRPC and GraphQL and adding a visualisation of the whole stack (gRPC -> Indexing Framework -> RPC service) as another way for people to decide what they would use (if they already have an RPC service that talks to a DB, then they only need an indexing framework, if they already have an indexing framework, then they only need to talk to gRPC, etc etc).